python/ray/_private/node.py

```diff
@@ -177,9 +178,34 @@ def __init__(
 
         # Register the temp dir.
         if head:
-            # date including microsecond
-            date_str = datetime.datetime.today().strftime("%Y-%m-%d_%H-%M-%S_%f")
-            self._session_name = f"session_{date_str}_{os.getpid()}"
+            maybe_key = None
+            if self._ray_params.external_addresses is not None:
+                self._redis_address = self._ray_params.external_addresses[0]
+                parts = self._redis_address.split("://", 1)
+                enable_redis_ssl = False
+                if len(parts) == 1:
+                    redis_ip_address, redis_port = parts[0].rsplit(":", 1)
+                else:
+                    if len(parts) != 2 or parts[0] not in ("redis", "rediss"):
+                        raise ValueError(f"Invalid redis address {self._redis_address}")
+                    redis_ip_address, redis_port = parts[1].rsplit(":", 1)
+                    if parts[0] == "rediss":
+                        enable_redis_ssl = True
+                maybe_key = get_key_from_storage(
```

Comment:
Can you add the same validation logic as `cleanup_redis_storage`?

Reply:
Added it for ports because the other type checks will happen during runtime and yield a very helpful error anyway (at least that was my experience with pytest on a type mismatch here).